channelmonitor/channelmonitor.go

```diff
@@ -187,8 +187,33 @@ func newMonitoredChannel(
 		cfg:        cfg,
 		onShutdown: onShutdown,
 	}
+
+	// "debounce" calls to restart channel, ie if there are multiple calls in a
+	// short space of time, only send a message to restart the channel once
+	var lk sync.Mutex
+	var lastErr error
 	debouncer := debounce.New(cfg.RestartDebounce)
-	mpc.restartChannelDebounced = func() { debouncer(mpc.restartChannel) }
+	mpc.restartChannelDebounced = func(err error) {
+		// Log the error at debug level
+		log.Debug(err.Error())
```

Comment:
FYI:

```go
log.Debugw("what does this mean", "error", err)
```

Otherwise:

1. We pay for the overhead of calling `Error`, even if we don't have debug logging enabled.
2. The user will see log messages like "stream reset" with no context.